command_line/index.py

```diff
@@ -119,106 +122,81 @@ def index_experiments(experiments, reflections, params, known_crystal_models=Non
     return idxr.refined_experiments, idx_refl
 
 
-class Index(object):
-    def __init__(self, experiments, reflections, params):
-
-        self._params = params
-
-        if experiments.crystals()[0] is not None:
-            known_crystal_models = experiments.crystals()
-        else:
-            known_crystal_models = None
-
-        if len(reflections) == 0:
-            raise Sorry("No reflection lists found in input")
-        elif len(reflections) == 1:
-            reflections[0]["imageset_id"] = reflections[0]["id"]
-        elif len(reflections) > 1:
-            assert len(reflections) == len(experiments)
-            for i in range(len(reflections)):
-                reflections[i]["imageset_id"] = flex.int(len(reflections[i]), i)
-                if i > 0:
-                    reflections[0].extend(reflections[i])
-        reflections = reflections[0]
-
-        for expt in experiments:
-            if (
-                expt.goniometer is not None
-                and expt.scan is not None
-                and expt.scan.get_oscillation()[1] == 0
-            ):
-                expt.goniometer = None
-                expt.scan = None
-
-        if self._params.indexing.image_range:
-            reflections = slice_reflections(
-                reflections, self._params.indexing.image_range
-            )
-
-        if len(experiments) == 1 or self._params.indexing.joint_indexing:
-            try:
-                self._indexed_experiments, self._indexed_reflections = index_experiments(
-                    experiments,
-                    reflections,
-                    copy.deepcopy(params),
-                    known_crystal_models=known_crystal_models,
-                )
-            except DialsIndexError as e:
-                raise Sorry(str(e))
-        else:
-            self._indexed_experiments = ExperimentList()
-            self._indexed_reflections = flex.reflection_table()
-
-            import concurrent.futures
-
-            with concurrent.futures.ProcessPoolExecutor(
-                max_workers=params.indexing.nproc
-            ) as pool:
-                futures = []
-                for i_expt, expt in enumerate(experiments):
-                    refl = reflections.select(reflections["imageset_id"] == i_expt)
-                    refl["imageset_id"] = flex.size_t(len(refl), 0)
-                    futures.append(
-                        pool.submit(
-                            index_experiments,
-                            ExperimentList([expt]),
-                            refl,
-                            copy.deepcopy(params),
-                            known_crystal_models=known_crystal_models,
-                        )
+def index(experiments, reflections, params):
```

Comment:
DIALS and xia2 are riddled with functions called `index`, `indexer`, etc..  Here, it's definitely appropriate, but to make it clear what _this_ function does (it's the main one after all), I think it would be quite handy to have a docstring.